include/stl2/detail/algorithm/sort.hpp

```diff
@@ -24,81 +24,125 @@
+		requires Sortable<I, Comp, Proj>
+		static constexpr void introsort_loop(I first, I last, iter_difference_t<I> depth_limit,
+			Comp& comp, Proj& proj)
+		{
+			while (distance(first, last) > introsort_threshold) {
+				if (depth_limit == 0) {
+					partial_sort(first, last, last, std::ref(comp), std::ref(proj));
+					return;
+				}
+				I cut = __sort_fn::unguarded_partition(first, last, comp, proj);
+				__sort_fn::introsort_loop(cut, last, --depth_limit, comp, proj);
+				last = cut;
+			}
+		}
+
+		template<BidirectionalIterator I, class Comp, class Proj>
+		requires Sortable<I, Comp, Proj>
+		static constexpr void unguarded_insertion_sort(I first, I last, Comp& comp, Proj& proj)
+		{
+			for (I i = first; i != last; ++i) {
+				detail::rsort::unguarded_linear_insert(i, __stl2::iter_move(i), comp, proj);
+			}
 		}
-		auto last = __stl2::next(first, std::move(sent));
-		auto n = __stl2::distance(first, last);
-		detail::rsort::introsort_loop(first, last, detail::rsort::log2(n) * 2, comp, proj);
-		detail::rsort::final_insertion_sort(first, last, comp, proj);
-		return last;
-	}
 
-	template<RandomAccessRange Rng, class Comp = less, class Proj = identity>
-	requires
-		Sortable<iterator_t<Rng>, Comp, Proj>
-	safe_iterator_t<Rng> sort(Rng&& rng, Comp comp = {}, Proj proj = {})
-	{
-		return __stl2::sort(begin(rng), end(rng),
-			std::ref(comp), std::ref(proj));
-	}
+		template<RandomAccessIterator I, class Comp, class Proj>
+		requires Sortable<I, Comp, Proj>
+		static constexpr void final_insertion_sort(I first, I last, Comp &comp, Proj &proj)
+		{
+			if (distance(first, last) > introsort_threshold) {
+				detail::rsort::insertion_sort(first, first + introsort_threshold, comp, proj);
+				__sort_fn::unguarded_insertion_sort(first + introsort_threshold, last, comp, proj);
+			} else {
+				detail::rsort::insertion_sort(first, last, comp, proj);
+			}
+		}
 
-	namespace ext {
-#if STL2_WORKAROUND_GCC_79591
-		template<RandomAccessIterator I, Sentinel<I> S, class Comp = less,
-			class Proj = identity>
-		requires
-			Sortable<I, Comp, Proj>
-		I sort(I first, S sent, Comp comp = {}, Proj proj = {})
+		template <Integral I>
+		static constexpr auto log2(I n) {
+			STL2_EXPECT(n > 0);
+			I k = 0;
+			for (; n != 1; n /= 2) {
+				++k;
+			}
+			return k;
+		}
+	public:
+		template<RandomAccessIterator I, Sentinel<I> S, class Comp = less, class Proj = identity>
+		requires Sortable<I, Comp, Proj>
+		constexpr I operator()(I first, S sent, Comp comp = {}, Proj proj = {}) const
 		{
 			if (first == sent) {
 				return first;
 			}
-			auto last = __stl2::next(first, std::move(sent));
-			auto n = __stl2::distance(first, last);
-			detail::rsort::introsort_loop(first, last, detail::rsort::log2(n) * 2, comp, proj);
-			detail::rsort::final_insertion_sort(first, last, comp, proj);
+			auto last = next(first, std::move(sent));
+			auto n = distance(first, last);
+			__sort_fn::introsort_loop(first, last, __sort_fn::log2(n) * 2, comp, proj);
```

Comment:
We don't need to qualify calls to member functions of the same class ([basic.lookup.unqual]/1 and /8).
```suggestion
			introsort_loop(first, last, log2(n) * 2, comp, proj);
```

Reply:
![image](https://user-images.githubusercontent.com/8622748/49901013-0dfc0f00-fe58-11e8-85f7-d87691b93053.png)